avocado/core/runner.py

```diff
@@ -233,21 +228,13 @@ def _run_test(self, test_factory, queue):
         sys.stdout = output.LoggingFile(logger=logger_list_stdout)
         sys.stderr = output.LoggingFile(logger=logger_list_stderr)
 
-        try:
-            instance = loader.load_test(test_factory)
-            if instance.runner_queue is None:
-                instance.runner_queue = queue
-            runtime.CURRENT_TEST = instance
-            early_state = instance.get_state()
-            early_state['early_status'] = True
-            queue.put(early_state)
-        except Exception:
```

Comment:
Actually I thought about it and it's a corner case, the main thread should cope with that (using the 10s deadline) so yes, it's probably time to remove this hack and simply run the test. (the simpler - the better)